### PR TITLE
Add global-workflow infrastructure for ocean analysis recentering task

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
# Description

Adds jjob, rocoto script, config file, and basic `config.resources` entry for ocean analysis recentering task

This PR is a dependency for further work on the associated issue within global-workflow and GDASApp 

Refs https://github.com/NOAA-EMC/GDASApp/issues/912

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO (By itself it will not change functionality within global-workflow, but is necessary to change functionality within GDASApp)
- Does this change require a documentation update? NO

# How has this been tested?

Tested with GDASApp SOCA ctests with branch `feature/ocnanalecen`

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
FYI @guillaumevernieres

File: parm/config/gfs/config.resources [hunk 1]
```diff
@@ -400,6 +401,34 @@ case ${step} in
...
+        memory_ocnanalecen="96GB"
+        ;;
+      "C48")
+        npes=16
+        memory_ocnanalecen="24GB"
+        ;;
+      *)
+        echo "FATAL ERROR: Resources not defined for job ${job} at resolution ${CASE}"
+        exit 4
+    esac
```
Comment (aerorahul) [inline parm/config/gfs/config.resources]: Is the ocean recentering job dependent on atmosphere resolution?
Probably should have caught this on the other `ocn` jobs as well.

Comment (guillaumevernieres) [reply parm/config/gfs/config.resources]: Ha! Good point. No it isn't, we didn't upgrade the scripting after the ocean resolution was added as an option.

Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.resources]: What should it be? The entries for the ocean all key off of the atmospheric resolution.

Comment (guillaumevernieres) [reply parm/config/gfs/config.resources]: > What should it be? The entries for the ocean all key off of the atmospheric resolution.

@AndrewEichmann-NOAA , the variable is `OCNRES`, and I think the cases are `0.25`, `1.00` and `5.00`. That or x100, you'll have to check ...

Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.resources]: Ok should be fixed @guillaumevernieres @aerorahul

File: parm/config/gfs/config.resources [hunk 2]
```diff
@@ -357,12 +357,12 @@ case ${step} in
 
   "ocnanalbmat")
     npes=16
-    case ${CASE} in
-      "C384") npes=480;;
-      "C96")  npes=16;;
-      "C48")  npes=16;;
+    case ${OCNRES} in
```
Comment (aerorahul) [inline parm/config/gfs/config.resources:359]: I don't think 5-deg case needs 480 pes and the 0.25-deg case runs on 16.

File: parm/config/gfs/config.resources (hunk 1 above)
Comment (guillaumevernieres) [reply parm/config/gfs/config.resources]: looks like you've got the resource allocation for the resolutions backward @AndrewEichmann-NOAA .

Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.resources]: @guillaumevernieres d'oh

File: parm/config/gfs/config.resources (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.resources:359]: now unreversed

**Review by aerorahul — APPROVED** [review]
thanks.

**Review by guillaumevernieres — APPROVED** [review]

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+##############################################
+# Set variables used in the script
+##############################################
+
+##############################################
+# Begin JOB SPECIFIC work
+##############################################
+
+export PYTHONPATH=${HOMEgfs}/sorc/gdas.cd/ush:${PYTHONPATH}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: This does not belong in the j-job.  It should be in the rocoto job.  See example for other JEDI tasks

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+###############################################################
+# Run relevant script
+
+EXSCRIPT=${GDASOCNRUNSH:-${HOMEgfs}/sorc/gdas.cd/scripts/exgdas_global_marine_analysis_ecen.py}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: This is not a shell script.
Also, `GDASOCNRUNSH` is **not** the variable for this script.  It is used in the `ocnanalrun`

**Review by aerorahul — CHANGES_REQUESTED** [review]
some comments on the job scripts.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 3 above)
Comment (CoryMartin-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: Yes, this should mirror the aero jobs for example

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 4 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: oof, changed.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 3 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: moved

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+source "${HOMEgfs}/ush/preamble.sh"
+export WIPE_DATA="NO"
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: Is this job supposed to reuse this DATA directory?

Comment (guillaumevernieres) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: No, that's a left over copy/paste, needs to be corrected @AndrewEichmann-NOAA .

Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: fixed

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalecen" -c "base ocnanal ocnanalecen"
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN:3]: why is `ocnanal` config needed in this task?

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+##############################################
+
+##############################################
+# Begin JOB SPECIFIC work
+##############################################
+
+###############################################################
+# Run relevant script
+
+EXSCRIPT=${GDASOCNCENPY:-${HOMEgfs}/ush/exgdas_global_marine_analysis_ecen.py}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: this is an exscript, and should be in `scripts/` not `ush/`

File: jobs/rocoto/ocnanalecen.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+. "${HOMEgfs}/ush/load_ufsda_modules.sh"
+status=$?
+[[ ${status} -ne 0 ]] && exit "${status}"
+
+export job="ocnanalecen"
+export jobid="${job}.$$"
+
+###############################################################
+# Setup Python path for GDASApp ush
+PYTHONPATH=${HOMEgfs}/sorc/gdas.cd/ush:${PYTHONPATH}
```
Comment (aerorahul) [inline jobs/rocoto/ocnanalecen.sh]: If `PYTHONPATH` is not defined before this point, the `PYTHONPATH` set after this line will end in a `:`.  This should be rectified.

**Review by aerorahul — CHANGES_REQUESTED** [review]

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 6 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN:3]: For `JEDI_BIN`

Comment (aerorahul) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN:3]: What is in `JEDI_BIN`? All executables are in `HOMEgfs/exec`

File: jobs/rocoto/ocnanalecen.sh (hunk 8 above)
Comment (AndrewEichmann-NOAA) [reply jobs/rocoto/ocnanalecen.sh]: fixed

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 7 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN]: fixed

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN (hunk 6 above)
Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_ECEN:3]: Also for `SOCA_INPUT_FIX_DIR`

**Comment by AndrewEichmann-NOAA** [comment]
@aerorahul Does this need any more work? We'll need this soon for runs

File: env/HERA.env [hunk 9]
```diff
@@ -107,6 +107,12 @@ elif [[ "${step}" = "ocnanalchkpt" ]]; then
 
     export APRUN_OCNANAL="${launcher} -n ${npe_ocnanalchkpt}"
 
+elif [[ "${step}" = "ocnanalecen" ]]; then
+
+    export APRUNCFP="${launcher} -n \$ncmd --multi-prog"
```
Comment (guillaumevernieres) [inline env/HERA.env]: `APRUNCFP` is not needed

**Review by guillaumevernieres — COMMENTED** [review]
You need to also update ORION.env @AndrewEichmann-NOAA

File: env/HERA.env [hunk 10]
```diff
@@ -107,6 +107,12 @@ elif [[ "${step}" = "ocnanalchkpt" ]]; then
...
+
+    export APRUN_OCNANALECEN="${launcher} -n ${npe_ocnanalecen}"
```
Comment (aerorahul) [inline env/HERA.env]: I think we need to add the `--cpus-per-task` argument to prevent oversubscribing

File: env/HERA.env (hunk 9 above)
Comment (AndrewEichmann-NOAA) [reply env/HERA.env]: removed

File: env/HERA.env (hunk 10 above)
Comment (AndrewEichmann-NOAA) [reply env/HERA.env]: There apparent is some issue with this that can turn up at higher resolutions - can we wait to see if we need this?

**Review by aerorahul — APPROVED** [review]

**Pull request merged**